/*
 * netsiege - A multiplayer point and click adventure
 * Copyright (C) 2014 Gregor Vollmer <gregor@celement.de>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package network.pb;
import "src/oms/objects.proto";


message RPC {
	required uint32 msg_seq_id=1;
	required uint32 rpc_id=2;
	required string rpc_data=3;
}

message S2CMessage {
	message JoinServerResponse {
		required uint32 player_id=1;
		required uint32 server_version=2;
		required bool player_accepted=3;
		required string map_name=4;
		required string map_version=5;
	}
	message WorldStateUpdate {
		required omsproto.GameObjectSet updated_objects=1;
		repeated uint32 removed_ids=2;
	}
	required uint32 protocol_id=1;
	required uint32 seq_id=2;
	required uint32 ack_seq_id=3;
	required uint32 ack_mask=4;
	optional JoinServerResponse join_server_response=5;
	optional WorldStateUpdate world_state_update=6;
	optional RPC rpc=7;
}

message C2SMessage {
	message JoinServerReq {
		required uint32 client_version=1;
		required string name=2;
	}
	required uint32 protocol_id=1;
	required uint32 seq_id=2;
	required uint32 ack_seq_id=3;
	required uint32 ack_mask=4;
	optional RPC rpc=5;
	optional JoinServerReq join_server_req=6;
}
